std/assembly/string.ts

```diff
@@ -438,6 +438,52 @@ export class String {
     return out;
   }
 
+  padStart(targetLength: i32, padString: String = changetype<String>(" ")): String {
+    assert(this !== null);
+    var length = this.length;
+    var padLen = padString.length;
+    if (targetLength < length || !padLen) return this;
+    var len = targetLength - length;
+    if (len > padLen) padString = padString.repeat((len + padLen - 1) / padLen);
+    var out = allocate(targetLength);
+    memory.copy(
+      changetype<usize>(out) + HEADER_SIZE,
+      changetype<usize>(padString) + HEADER_SIZE,
+      <usize>len << 1
+    );
+    if (length) {
+      memory.copy(
+        changetype<usize>(out) + HEADER_SIZE + (<usize>len << 1),
+        changetype<usize>(this) + HEADER_SIZE,
+        <usize>targetLength << 1
+      );
+    }
+    return out;
+  }
+
+  padEnd(targetLength: i32, padString: String = changetype<String>(" ")): String {
+    assert(this !== null);
+    var length = this.length;
+    var padLen = padString.length;
+    if (targetLength < length || !padLen) return this;
+    var len = targetLength - length;
+    if (len > padLen) padString = padString.repeat((len + padLen - 1) / padLen);
+    var out = allocate(targetLength);
+    if (length) {
+      memory.copy(
+        changetype<usize>(out) + HEADER_SIZE,
+        changetype<usize>(this) + HEADER_SIZE,
+        <usize>length << 1
+      );
+    }
+    memory.copy(
+      changetype<usize>(out) + HEADER_SIZE + (<usize>length << 1),
+      changetype<usize>(padString) + HEADER_SIZE,
+      <usize>targetLength << 1
```

Comment:
Does this copy out of bounds? Rather `<usize>len << 1` ?

Reply:
Fixed!